### DISCUSSION TITLE
refetchOnMount: 'always' does not purge the cache if staleTime is set

### DISCUSSION THREAD

**Discussion by Zerebokep** [TanStack/query#4377] [Q&A]
### Describe the bug

As described in the title.

### Your minimal, reproducible example

-

### Steps to reproduce

-

### Expected behavior

It should always refetch, no matter if staleTime or cacheTime is set.

### How often does this bug happen?

Every time

### Platform

-

### react-query version

-

**Comment by Zerebokep** [comment]
Thanks.

**Comment by TkDodo** [comment]
on a more serious note: There's a reason why all fields are mandatory, like the reproduction one. I'll move this to a discussion because I think there's just a misunderstanding about how you think things work and how they really work, so you need to more specific with your question instead of just opening an issue with barely any information in it. What did you think that would achieve?

> refetchOnMount: 'always' does not purge the cache if staleTime is set

- nothing in react-query will "purge the cache", so this also doesn't.
- `staleTime` plays no role for `refetchOnMount` if it is set to `always`.

If you set `refetchOnMount` to `always`, you will get a background refetch even if your data is not stale. That's all. stale data will always be returned if there is any in the cache because that's how react-query works.

**Comment by BlindMonkLeeSin** [comment]
I had the same problem as him

When remounting, if the data is fresh, then refetchOnMount is set to 'always' and does not refresh the data

**Reply by TkDodo** [reply]
then show a reproduction please
